fpop/prep_fp.py

```diff
@@ -0,0 +1,178 @@
+from abc import ABC,abstractmethod
+import os
+import dpdata
+from contextlib import contextmanager
+from pathlib import Path
+from dflow.python import (
+    PythonOPTemplate,
+    OP,
+    OPIO,
+    OPIOSign,
+    Artifact,
+    TransientError,
+    FatalError,
+    BigParameter,
+)
+from typing import (
+    Any,
+    Tuple,
+    List,
+    Set,
+    Dict,
+    Optional,
+    Union,
+)
+
+@contextmanager
+def set_directory(path: Path):
+    '''Sets the current working path within the context.
+    Parameters
+    ----------
+    path : Path
+        The path to the cwd
+    Yields
+    ------
+    None
+    Examples
+    --------
+    >>> with set_directory("some_path"):
+    ...    do_something()
+    '''
+    cwd = Path().absolute()
+    path.mkdir(exist_ok=True, parents=True)
+    try:
+        os.chdir(path)
+        yield
+    finally:
+        os.chdir(cwd)
```

Comment:
Use the tools `set_directory` in `dflow`

Reply:
I have done it.